lib/optimizely/event/forwarding_event_processor.rb

```diff
@@ -30,11 +30,14 @@ def process(user_event)
       log_event = Optimizely::EventFactory.create_log_event(user_event, @logger)
 
       begin
-        @event_dispatcher.dispatch_event(log_event)
-        @notification_center&.send_notifications(
-          NotificationCenter::NOTIFICATION_TYPES[:LOG_EVENT],
-          log_event
-        )
+        @notification_center&.send_notifications(NotificationCenter::NOTIFICATION_TYPES[:LOG_EVENT], log_event)
```

Comment:
Shouldn't we be only firing notification when the event has successfully dispatched?

Reply:
Also, why not also fire notification in the thread as well. this will allow us to quickly free up our main thread if the client has registered some time-taking callback in the LogEvent listener. And, if we do this, we can simply wait for the callback to get called in our unit tests rather than putting it to sleep before assertions.

Reply:
Yeah, I was debating back and forth on that.  Originally, it was in the thread.  However, all the other SDKs fire the event before dispatch.  So, I thought this was ok.  I can put it back in the thread.  it might be better for the notification listener to not be on the main thread.